File: Python/01/src/game.py
import discord
from discord_together import DiscordTogether
from discord.ext import commands
import random
from asyncio.exceptions import TimeoutError as ast
from config import config

words = ["tennis","badminton","nice","cars","volleyball"]

class game(commands.Cog):
    def __init__(self, client):
        self.client = client

    @commands.Cog.listener()
    async def on_ready(self):
        self.togetherControl = await DiscordTogether(config.token)

    @commands.command(aliases=["ytt","youtube"])
    async def YouTube_Together(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "youtube")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Youtube Together** แล้วค่ะ!\n{link}")

    @commands.command(aliases=["poker"])
    async def Poker_Night(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "poker")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Poker Night** แล้วค่ะ!\n{link}")

    @commands.command(aliases=["chess"])
    async def Chess_in_the_Park(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "chess")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Chess in the Park** แล้วค่ะ!\n{link}")

    @commands.command(aliases=["betrayal"])
    async def Betrayal_io(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "betrayal")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Betrayal.io** แล้วค่ะ!\n{link}")

    @commands.command(aliases=["fishing"])
    async def Fishington_io(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "fishing")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Fishington.io** แล้วค่ะ!\n{link}")

    @commands.command(aliases=["letter-tile"])
    async def Letter_Tile(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "letter-tile")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Letter Tile** แล้วค่ะ!\n{link}")

    @commands.command(aliases=["word-snack","word"])
    async def Word_Snack(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "word-snack")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Word Snack** แล้วค่ะ!\n{link}")

    @commands.command(aliases=["doodle-crew"])
    async def Doodle_Crew(self, ctx):
        try:
            ctx.author.voice.channel
        except AttributeError:
            return await cant(self.client, ctx)
        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, "doodle-crew")
        await ctx.reply(f"`{ctx.author}` ได้สร้างกิจกรรม **Doodle Crew** แล้วค่ะ!\n{link}")

    @commands.command()
    async def hangman(self, ctx):
        word = random.choice(words)
        all_char = list(word)
        message = await ctx.reply(embed=discord.Embed(
            title=gen_display(word, all_char),
            description=f"จำนวนครั้งผิด `{0}`",
            color=0x00ffff
        ).set_author(
            name="Hangman Game",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))
        try:
            _input = await self.client.wait_for('message', timeout=30)
            await play(self, message, _input, all_char, word)
        except ast:
            await message.delete()
            return await message.edit(embed=discord.Embed(
                title="หมดเวลาแล้วค่ะ\n" + gen_display(word, all_char),
                description=f"จำนวนครั้งผิด `{0}`",
                color=0x00ffff
            ).set_author(
                name="Hangman Game",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))

def setup(client):
    client.add_cog(game(client))

async def play(self, message, _input, all_char, word, cor_count=0, incor_count=0):
    if _input.content in all_char:
        cor_count += 1
        for n, i in enumerate(all_char):
            if i == _input.content :
                all_char[n] = "` `"
                break
        await _input.delete()
        if cor_count == len(all_char): return await message.edit(embed=discord.Embed(
            title="คุณชนะแล้วค่ะ!\n" + gen_display(word, all_char),
            description=f"จำนวนครั้งผิด `{incor_count}`",
            color=0x00ffff
        ).set_author(
            name="Hangman Game",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))
    else:
        incor_count += 1
        await _input.delete()
    await message.edit(embed=discord.Embed(
            title=gen_display(word, all_char),
            description=f"จำนวนครั้งผิด `{incor_count}`",
            color=0x00ffff
        ).set_author(
            name="Hangman Game",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))
    try:
        _input = await self.client.wait_for('message', timeout=30)
    except ast:
        return await message.edit(embed=discord.Embed(
            title="หมดเวลาแล้วค่ะ\n" + gen_display(word, all_char),
            description=f"จำนวนครั้งผิด `{incor_count}`",
            color=0x00ffff
        ).set_author(
            name="Hangman Game",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))
    await play(self, message, _input, all_char, word, cor_count, incor_count)

def gen_display(word, all_char):
    gen = []
    for i in range(len(list(word))):
        if list(word)[i] == all_char[i]:
            gen.append("` `")
            gen.append(" ")
        else:
            gen.append("`" + list(word)[i] + "`")
            gen.append(" ")
    return "".join(gen)

async def cant(client, ctx):
    await ctx.reply(embed=discord.Embed(
        title=f"ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
        description="โปรดเข้าช่องเสียงก่อน เพื่อใช้คำสั่งนี้นะคะ",
        color=0x00ffff
    ).set_author(
        name="ไม่สามารถดำเนินการได้ค่ะ!",
        icon_url=client.user.avatar_url,
        url=config.author_url
    ))

File: JS/01/src/commands/game.js
const { MessageEmbed } = require('discord.js')
const request = require('request')

module.exports = [
    {
        name: "youtube",
        aliases: ["ytt","youtube-together"],
        category: "game",
        information: "สร้าง activity discord - Youtube Together",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Youtube Together** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "880218394199220334")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "poker",
        category: "game",
        information: "สร้าง activity discord - Poker Night",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Poker Night** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "755827207812677713")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "betrayal",
        category: "game",
        information: "สร้าง activity discord - Betrayal io",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Betrayal io** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "773336526917861400")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "fishing",
        category: "game",
        information: "สร้าง activity discord - Fishington io",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Fishington io** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "814288819477020702")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "chess",
        category: "game",
        information: "สร้าง activity discord - Chess in the Park",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Chess in the Park** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "832012774040141894")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "letter-tile",
        category: "game",
        information: "สร้าง activity discord - Letter Tile",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Letter Tile** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "879863686565621790")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "word-snack",
        category: "game",
        information: "สร้าง activity discord - Word Snack",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Word Snack** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "879863976006127627")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "doodle-crew",
        category: "game",
        information: "สร้าง activity discord - Doodle Crew",
        async run(client, message) {
            try {
                const { channel } = message.member.voice
                if (!channel) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบช่องเสียงที่คุณอยู่ค่ะ",
                        description: "คุณต้องอยู่ในช่องเสียงก่อน จึงจะสามารถสร้างได้นะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.loading.edit({content: `<@!${message.author.id}> ได้สร้างกิจกรรม **Doodle Crew** แล้วค่ะ!\n${await gen_activity_link(client, channel.id, "878067389634314250")}`, embeds:[]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
]

function gen_activity_link(client, channel_id, game_id) {
    return new Promise((resolve, reject) => {
        request.post({
            url: `https://discord.com/api/v9/channels/${channel_id}/invites`,
            headers: {
                "Authorization": `Bot ${client.config.token}`,
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                max_age: 0,
                max_uses: 0,
                target_application_id: game_id,
                target_type: 2,
                temporary: false,
                validate: null,
            })
        }, (err, res) => {
            if (err) reject("[err] request error.", err)
            if (!res.body) reject("[err] request error, response is don't body.")
            resolve("https://discord.gg/" + JSON.parse(res.body).code)
        })
    })
}